src/hotspot/share/opto/compile.cpp

```diff
@@ -5370,3 +5370,143 @@ Node* Compile::narrow_value(BasicType bt, Node* value, const Type* type, PhaseGV
 void Compile::record_method_not_compilable_oom() {
   record_method_not_compilable(CompilationMemoryStatistic::failure_reason_memlimit());
 }
+
+#ifndef PRODUCT
+// Collects all the control inputs from nodes on the worklist and from their data dependencies
+static void find_candidate_control_inputs(Unique_Node_List& worklist, Unique_Node_List& candidates) {
+  // Follow non-control edges until we reach CFG nodes
+  for (uint i = 0; i < worklist.size(); i++) {
+    const Node* n = worklist.at(i);
+    for (uint j = 0; j < n->req(); j++) {
+      Node* in = n->in(j);
+      if (in == nullptr || in->is_Root()) {
+        continue;
+      }
+      if (in->is_CFG()) {
+        if (in->is_Multi()) {
+          candidates.push(in->as_Multi()->proj_out(TypeFunc::Control));
```

Comment:
Why you collect only one control output edge? IfNode has two. May be add comment if it is intentional.

Reply:
Then which CFG/Multi nodes you expect here?

Reply:
I suggest to add `assert(!n->is_MultiBranch(), "unexpected node: %s", n->Name());`  if you are sure we never see such nodes here.

Reply:
Done, thanks for the suggestion!